### PR TITLE
#comment Resolve Photometric_Interpretations for MONOCHROME2 and MONO…

### PR DESCRIPTION
**Pull request by jgonzabal**
…CHROME1 inverting grayscale.

### CODE REVIEW

**Comment by jgonzabal** [comment]
This must work.

File: src/parsers/parsers.dicom.js [hunk 1]
```diff
@@ -873,37 +691,6 @@ export default class ParsersDicom extends ParsersVolume {
 
     return rgbData;
   }
-
-  /**
-   * Swap bytes in frame.
-   */
-  _swapFrame(frame) {
```
Comment (NicolasRannou) [inline src/parsers/parsers.dicom.js]: We need this method to handle big endians

File: src/parsers/parsers.dicom.js [hunk 2]
```diff
@@ -553,74 +424,47 @@ export default class ParsersDicom extends ParsersVolume {
...
       transferSyntaxUID === '1.2.840.10008.1.2.4.51') {        // JPEG Baseline lossy process 2 & 4 (12 bit)
-
       // JPEG Baseline
       return this._decodeJPEGBaseline(frameIndex);
-
     } else if (transferSyntaxUID === '1.2.840.10008.1.2' || // Implicit VR Little Endian
-      transferSyntaxUID === '1.2.840.10008.1.2.1') {      // Explicit VR Little Endian
-
-      // get data
+      transferSyntaxUID === '1.2.840.10008.1.2.1' ||     // Explicit VR Little Endian
```
Comment (NicolasRannou) [inline src/parsers/parsers.dicom.js]: we need 2 cases for little and bug endians

File: src/helpers/helpers.slice.js [hunk 3]
```diff
@@ -225,6 +226,12 @@ export default class HelpersSlice extends THREE.Object3D{
       this._center = this._stack.centerAABBox();
       this._toAABB = this._stack.lps2AABB;
     }
+
```
Comment (NicolasRannou) [inline src/helpers/helpers.slice.js]: no need anymore

File: src/helpers/helpers.slice.js [hunk 4]
```diff
@@ -22,6 +22,7 @@ export default class HelpersSlice extends THREE.Object3D{
     // image settings
     // index only used to grab window/level and intercept/slope
     this._invert = false;
```
Comment (NicolasRannou) [inline src/helpers/helpers.slice.js]: = this._stack.invert should be working there now

File: src/helpers/helpers.slice.js [hunk 5]
```diff
@@ -227,11 +227,7 @@ export default class HelpersSlice extends THREE.Object3D{
       this._toAABB = this._stack.lps2AABB;
     }
 
-    try {
-      this._invert = this._stack.invert;
-    } catch (err) {
-      this._invert = false;
-    }
+    this._invert = this._stack.invert;
```
Comment (NicolasRannou) [inline src/helpers/helpers.slice.js]: Can we do it directly line 24 in the constructor?

**Comment by NicolasRannou** [comment]
thanks @jgonzabal !

**Pull request merged**